platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/cli/BuildActionsFactory.java

```diff
@@ -217,15 +218,16 @@ private Runnable runBuildInProcess(StartParameterInternal startParameter, Daemon
         ServiceRegistry globalServices = buildProcessState.getServices();
         globalServices.get(AgentInitializer.class).maybeConfigureInstrumentationAgent();
 
-        BuildActionExecutor<BuildActionParameters, BuildRequestContext> executer = new ForwardStdInToThisProcess(
-            globalServices.get(GlobalUserInputReceiver.class),
-            globalServices.get(UserInputReader.class),
-            System.in,
-            globalServices.get(BuildExecutor.class)
-        );
+        BuildActionExecutor<BuildActionParameters, ClientBuildRequestContext> executor = new RunInProcess(
```

Comment:
Use the client-side context instead of the daemon-side context in the `gradle` client, but when running in-process, use some adapters to convert client-side to daemon-side.